src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -347,8 +353,18 @@ public void snapshotCreateBQDataset(BigQueryProject bigQueryProject, Snapshot sn
         if (bigQueryProject.datasetExists(snapshotName)) {
             bigQueryProject.deleteDataset(snapshotName);
         }
+
+        // TODO: When we support multiple datasets per snapshot, this will need to be reworked
+        String representativeRegion = snapshot.getSnapshotSources()
+            .get(0)
+            .getDataset()
+            .getDatasetSummary()
+            .getStorage()
+            .stream()
+            .collect(Collectors.toMap(StorageResource::getCloudResource, StorageResource::getRegion))
+            .get(GoogleCloudResource.BIGQUERY.toString());
```

Comment:
This is the second place where we need to convert an enum to a string to look up a value. Could `StorageResource.getCloudResource()` return `GoogleCloudResource`? Or, could it have a boolean method `isBigQuery()`?